tests/validation/tests/v3_api/test_group_grb.py

```diff
@@ -1,110 +1,386 @@
-but is in a group that is assigned to an admin global role, has admin has
-admin privileges.
-
-Steps:
-1. login to user A who is in group B
-2. attempt to create a resources that requires admin privileges
-    a. assert Forbidden error
-3. create a globalrolebinding that assigns group B to the admin
-globalrole.
-    a. assert user A can now create a resource that requires admin
-    privileges
-4. delete globalrolebinding created in step 3.
-5. attempt to create a resource that requires admin privileges
-    a. assert Forbidden error
+    - Enable Auth
+    - Optional: searching nested group enabled
+    - All users used for testing global group role binding should not be used
+      to create the cluster when preparing the Rancher setup
 '''
 
-AUTH_PROVIDER = os.environ.get('RANCHER_AUTH_PROVIDER', "")
-
-# username of auth user that only has user privileges
-RANCHER_AUTH_USERNAME = os.environ.get('RANCHER_AUTH_USERNAME', "")
-# password for auth user
-RANCHER_AUTH_PASSWORD = os.environ.get('RANCHER_AUTH_PASSWORD', "")
-# name of auth group that auth user is a member of
-RANCHER_AUTH_GROUP = os.environ.get('RANCHER_AUTH_GROUP', "")
+namespace = {"auth_info": None}
+# values used to create a catalog
+BRANCH = "dev"
+URL = "https://git.rancher.io/system-charts"
 
+AUTH_PROVIDER = os.environ.get('RANCHER_AUTH_PROVIDER', "activeDirectory")
+NESTED_GROUP_ENABLED = ast.literal_eval(
+    os.environ.get('RANCHER_NESTED_GROUP_ENABLED', "False"))
+# the shared password for all auth users
+AUTH_USER_PASSWORD = os.environ.get('RANCHER_AUTH_USER_PASSWORD', "")
+# the link to log in as an auth user
 CATTLE_AUTH_URL = \
-    CATTLE_TEST_URL + \
-    "/v3-public/"+AUTH_PROVIDER+"Providers/" + \
-    AUTH_PROVIDER.lower()+"?action=login"
-
+    CATTLE_TEST_URL + "/v3-public/" \
+    + AUTH_PROVIDER + "Providers/" \
+    + AUTH_PROVIDER.lower() + "?action=login"
+# the link to enable or disable the auth
 CATTLE_AUTH_PROVIDER_URL = \
-    CATTLE_TEST_URL + "/v3/"+AUTH_PROVIDER+"Configs/"+AUTH_PROVIDER.lower()
-
+    CATTLE_TEST_URL + "/v3/" \
+    + AUTH_PROVIDER + "Configs/" \
+    + AUTH_PROVIDER.lower()
+# the link to search principals in the auth provider
 CATTLE_AUTH_PRINCIPAL_URL = CATTLE_TEST_URL + "/v3/principals?action=search"
 
 
-def test_group_grbs():
-    groups = search_ad_groups(RANCHER_AUTH_GROUP, ADMIN_TOKEN)
-    admin_client = get_admin_client()
+def test_ggrb_1(remove_resource):
+    """ test that when a global role is assigned to a group,
+    all users in the group will get the permission;
+    when the ggrb is removed, all users in the group will lose
+    the permission.
 
-    r = requests.post(CATTLE_AUTH_URL, json={
-        'username': RANCHER_AUTH_USERNAME,
-        'password': RANCHER_AUTH_PASSWORD,
-        'responseType': 'json',
-    }, verify=False)
+    the default global role "catalogs-manage" is used
+    """
 
-    token = r.json()["token"]
+    target_group_name = "nestgroup1"
```

Comment:
Can we have this target_group_name picked from the auth_info instead of hard coding it ? We can pick any group from the existing list of groups or if we are looking for some specific group we can have an additional parameter defined for it in the json file .

Reply:
updated